File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(arr, target) {
    const mpp=new Map();
    const n=arr.length;
    for(let i=0;i<n;i++){
        const num=arr[i];
        let remove=target-num;
        if(mpp.has(remove)){
            return [i,mpp.get(remove)];
        }
        mpp.set(num,i);
    }
    return [-1,-1];
};

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mpp;
        int n=nums.size();
        for(int i=0;i<n;i++){
            int remove=target-nums[i];
            if(mpp.find(remove)!=mpp.end()){
                return {mpp[remove],i};
            }
            mpp[nums[i]]=i;
        }
        return {-1,-1};
    }
};